### ISSUE TITLE
Incorrect logic inversion for expressions

### ISSUE THREAD

**Issue by julesbrx** [trullock/NUglify#160]
With default configutation, the filter "and" condition gets replaced by "or", and this changes the meaning of the filter :
```js
this.notifications.filter((n) => !n.Read && n.Type == menu.ID).length; // original
this.notifications.filter(n=>n.Read||n.Type==menu.ID).length // converted => not the same thing
```

I got around the issue by inverting the conditions :
```js
this.notifications.filter((n) => n.Type == menu.ID && !n.Read).length; // original
this.notifications.filter(n=>n.Type==menu.ID&&!n.Read).length // converted => OK
```

**Comment by trullock** [comment]
You're on it today! Will fix asap

**Comment by trullock** [comment]
fixed in 1.9.7
